app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,65 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
+
+  def new; end
+
+  def create
+    @user = User.find_by email:
+      params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "reset_pass.info"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "reset_pass.danger"
+      render :new
+    end
+  end
+
+  def update
+    if user_params[:password].blank?
+      flash.now[:fail] =
+        @user.errors.add(:password, t( "reset_pass.blank" ))
+      render :edit
+    elsif @user.update_attributes user_params
+      log_in @user
+      flash[:success] = t "reset_pass.success"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def edit; end
+
+  private
+
+  def get_user
+    @user = User.find_by email: params[:email]
+    unless @user
```

Comment:
Viết kiểu này sẽ gắn gọn và đẹp code hơn
```suggestion
    return if @user
    flash[:warning] = t "reset_pass.warning"
    redirect_to root_path
```

Reply:
ok anh.